src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -171,6 +171,10 @@ else if (!hasText(leftCurly)) {
                     ast.getText());
             }
         }
+
+        if (ast.getType() == TokenTypes.LITERAL_CASE) {
+            checkForEmptyCase(ast);
+        }
```

Comment:
why do you do separate processing ?
whole logic should be close to `if (emptyBlock) {`.
Please explain or refactor